src/ducks/stripe.duck.js

```diff
@@ -273,6 +273,154 @@ export const retrievePaymentIntentError = payload => ({
+      } else {
+        dispatch(handleCardPaymentSuccess(response));
+        return { ...response, transactionId };
+      }
+    })
+    .catch(err => {
+      // Unwrap Stripe error.
+      const e = err.error || storableError(err);
+      dispatch(handleCardPaymentError(e));
+
+      // Log error
+      const containsPaymentIntent = err.error && err.error.payment_intent;
+      const { code, doc_url, message, payment_intent } = containsPaymentIntent ? err.error : {};
+      const loggableError = containsPaymentIntent
+        ? {
+            code,
+            message,
+            doc_url,
+            paymentIntentStatus: payment_intent.status,
+          }
+        : e;
+      log.error(loggableError, 'stripe-handle-card-payment-failed', {
+        stripeMessage: loggableError.message,
+      });
+      throw e;
+    });
+};
+
+export const handleCardSetup = params => dispatch => {
+  // It's required to use the same instance of Stripe as where the card has been created
+  // so that's why Stripe needs to be passed here and we can't create a new instance.
+  const { stripe, card, setupIntentClientSecret, paymentParams } = params;
+
+  dispatch(handleCardSetupRequest());
+
+  return stripe
+    .handleCardSetup(setupIntentClientSecret, card, paymentParams)
+    .then(response => {
+      if (response.error) {
+        return Promise.reject(response);
+      } else {
+        dispatch(handleCardSetupSuccess(response));
+        return response;
+      }
+    })
+    .catch(err => {
+      // Unwrap Stripe error.
+      const e = err.error || storableError(err);
+      dispatch(handleCardSetupError(e));
+
+      // Log error
+      const containsSetupIntent = err.error && err.error.setup_intent;
+      const { code, doc_url, message, setup_intent } = containsSetupIntent ? err.error : {};
+      const loggableError = containsSetupIntent
+        ? {
+            code,
+            message,
+            doc_url,
+            paymentIntentStatus: setup_intent.status,
+          }
+        : e;
+      log.error(loggableError, 'stripe-handle-card-setup-failed', {
+        stripeMessage: loggableError.message,
+      });
+      throw e;
+    });
+};
+
+// TODO: when Create Stripe account API endpoint is updated, check that this function works correctly
+export const createStripeAccount = params => (dispatch, getState, sdk) => {
+  const country = params.country;
+  const bankAccountToken = params.bankAccountToken;
+
+  return sdk.stripeAccount.create({ country, bankAccountToken }, { expand: true }).catch(err => {
+    const e = storableError(err);
+    dispatch(stripeAccountCreateError(e));
+    const stripeMessage =
+      e.apiErrors && e.apiErrors.length > 0 && e.apiErrors[0].meta
+        ? e.apiErrors[0].meta.stripeMessage
+        : null;
+    log.error(err, 'create-stripe-account-failed', { stripeMessage });
+    throw e;
+  });
+};
+
+// ================ Deprecated thunk functions ================ //
```

Comment:
I wonder if we should comment these functions out - that way they are not in production build.
Another option would be to create some extra file: `stripeInAppOnboarding.duck.js`, that uses these functions and has alert on top of the file stating that we are not going to keep this Stripe API onboarding approach up-to-date - but focus on hosted onboarding.